src/test/java/bio/terra/workspace/service/resource/controlled/ControlledResourceServiceTest.java

```diff
@@ -140,23 +147,72 @@ public void createAiNotebookInstanceDo() throws Exception {
     assertEquals(
         FlightStatus.SUCCESS, stairwayComponent.get().getFlightState(jobId).getFlightStatus());
 
-    AIPlatformNotebooksCow notebooks = crlService.getAIPlatformNotebooksCow();
-    Instance instance =
-        notebooks
-            .instances()
-            .get(resource.toInstanceName(workspace.getGcpCloudContext().get().getGcpProjectId()))
-            .execute();
-    // git secrets gets a false positive if 'service_account' is double quoted.
-    assertThat(instance.getMetadata(), Matchers.hasEntry("proxy-mode", "service_" + "account"));
-    // TODO(PF-626): Test that the user has permission to act as the service account.
-
     assertEquals(
         resource,
         controlledResourceService.getControlledResource(
             workspace.getWorkspaceId(),
             resource.getResourceId(),
             userAccessUtils.defaultUserAuthRequest()));
 
+    // TODO(PF-643): this should happen inside WSM.
+    // Waiting 15s for permissions to propagate
+    TimeUnit.SECONDS.sleep(15);
+
+    InstanceName instanceName =
+        resource.toInstanceName(workspace.getGcpCloudContext().get().getGcpProjectId());
+    Instance instance =
+        crlService
+            .getAIPlatformNotebooksCow()
+            .instances()
+            .get(resource.toInstanceName(workspace.getGcpCloudContext().get().getGcpProjectId()))
+            .execute();
+
+    // Test that the user has permissions from WRITER roles on the notebooks instance. Only notebook
+    // instance level permissions can be checked on the notebook instance test IAM permissions
+    // endpoint, so no "notebooks.instances.list" permission as that's project level.
+    List<String> expectedWriterPermissions =
+        ImmutableList.of(
+            "notebooks.instances.get",
+            "notebooks.instances.reset",
+            "notebooks.instances.setAccelerator",
+            "notebooks.instances.setMachineType",
+            "notebooks.instances.start",
+            "notebooks.instances.stop",
+            "notebooks.instances.use");
+    assertThat(
```

Comment:
I'll opine some:
I don't think of the client vs connected test as about GCP vs internal WSM state, I think of it as testing the APIs at a coarser, higher level of user interaction (end-to-end) vs more detailed tests. As you move up abstraction layers, the tests integrate more parts, but have less detail. As demonstrated here, the GCP behavior, e.g. what happens when steps are retried, is interesting to test at this level where we have more control over execution and are closer to that code. 
A key area of WSM behavior is the GCP interaction, including setting roles and permissions. We can't meaningfully test that interaction as a unit test, since we need to use GCP to validate our GCP interaction. We can do that with "connected" tests, and therefore should do it there at the lowest level available to us. Our client test will exercise a simpler user journey, e.g. can I get a notebook, access it, delete it, but we don't need to write client tests for stop/start/set machine type if we're verifying in a connected test that those permissions are available. It's easier to get better test coverage at a lower testing layer.